feat: serverless worker (#4445)
temporalio/documentation: docs/evaluate/development-production-features/serverless-workers/index.mdx

---
id: index
title: Serverless Workers
sidebar_label: Serverless Workers
slug: /evaluate/serverless-workers
description:
  Understand the benefits of Serverless Workers and when to use them. Run Temporal Workers on serverless compute with no
  infrastructure to manage.
toc_max_heading_level: 4
keywords:
  - serverless
  - lambda
  - aws
  - worker
  - serverless worker
  - evaluate
tags:
  - Workers
  - Serverless
---

Serverless Workers let you run Temporal Workers on serverless compute platforms like AWS Lambda. There are no servers to
provision, no clusters to scale, and no idle compute to pay for. Temporal invokes the Worker when Tasks arrive, and the

> Worker shuts down when the work is done.

Reviewer: ```suggestion
Worker shuts down when the Tasks are done.
```

Serverless Workers use the same Temporal SDKs as traditional long-lived Workers. You register Workflows and Activities
the same way. The difference is in the lifecycle: instead of running a long-lived process, Temporal invokes the
Serverless Worker on demand when Tasks arrive. The Worker starts, polls for available Tasks, processes them, and exits

> when the work is done.

Reviewer: ```suggestion
when the Task is done.
```
